src/services/AdvancedTradingView/adapters/advancedTradingView.test.ts

```diff
@@ -0,0 +1,136 @@
+  chainId: ChainId.MAINNET,
+  decimals: 6,
+  name: 'USDC Coin',
+  symbol: 'USDC',
+} as Token
+
+const USDT_TOKEN = {
+  chainId: ChainId.MAINNET,
+  decimals: 6,
+  name: 'Tether Coin',
+  symbol: 'USDT',
+  address: '0xdAC17F958D2ee523a2206206994597C13D831ec7',
+} as Token
+
+const adapterConstructorParams = {
+  key: AdapterKeys.SWAPR,
+  adapterSupportedChains: [ChainId.MAINNET, ChainId.ARBITRUM_ONE, ChainId.GNOSIS],
+  platform: UniswapV2RoutablePlatform.SWAPR,
+  subgraphUrls: {
+    [ChainId.MAINNET]: 'https://api.thegraph.com/subgraphs/name/dxgraphs/swapr-mainnet-v2',
+    [ChainId.ARBITRUM_ONE]: 'https://api.thegraph.com/subgraphs/name/dxgraphs/swapr-arbitrum-one-v3',
+    [ChainId.GNOSIS]: 'https://api.thegraph.com/subgraphs/name/dxgraphs/swapr-xdai-v2',
+  },
+}
+
+const store = configureStore({
+  reducer: combineReducers({ advancedTradingView }),
+})
+
+describe('AdvancedTradingView - adapter', () => {
+  const advancedTradingViewAdapter = new AdvancedTradingViewAdapter({
+    adapters,
+    chainId: ChainId.MAINNET,
+    store: store as any,
+  })
+
+  it('adapters store is empty before fetching data', () => {
+    for (const adapterStore of Object.values(store.getState().advancedTradingView.adapters)) {
+      expect(!Object.keys(adapterStore).length).toBeTruthy()
+    }
+  })
+
+  it('isInitialized variable is equal to true after init method', () => {
+    expect(advancedTradingViewAdapter.isInitialized).toBeFalsy()
+
+    advancedTradingViewAdapter.init()
+
+    expect(advancedTradingViewAdapter.isInitialized).toBeTruthy()
+  })
+
+  it('setPairToken function set tokens inside store', () => {
+    expect(store.getState().advancedTradingView.pair.inputToken).toBe(undefined)
+    expect(store.getState().advancedTradingView.pair.outputToken).toBe(undefined)
+
+    advancedTradingViewAdapter.setPairTokens(USDC_TOKEN, USDT_TOKEN)
+
+    expect(store.getState().advancedTradingView.pair.inputToken).toBe(USDC_TOKEN)
+    expect(store.getState().advancedTradingView.pair.outputToken).toBe(USDT_TOKEN)
+  })
+})
+
+describe('BaseAdapter - swapr', () => {
+  it('check if adapter constructor was called', () => {
+    expect(BaseAdapter).not.toHaveBeenCalled()
+
+    new BaseAdapter(adapterConstructorParams)
+
+    expect(BaseAdapter).toHaveBeenCalledTimes(1)
+  })
+
+  it('getPairTrades fetch data and push it to store', async () => {
+    const baseAdapter = new BaseAdapter(adapterConstructorParams)
+
+    const fakeSwapData = {
+      id: '1',
+      transaction: {
+        id: '1',
+      },
+      amount0In: '10',
+      amount1In: '0',
+      amount0Out: '0',
+      amount1Out: '5',
+      amountUSD: '3',
+      timestamp: Date.now().toString(),
+    }
+
+    const fakePairId = '0x123456789'
+
+    jest.spyOn(baseAdapter, 'getPairTrades').mockImplementation(async () => {
+      const data = await new Promise<PairSwaps>(resolve => {
+        resolve({
+          swaps: [fakeSwapData],
+        })
+      })
+
+      store.dispatch(
```

Comment:
you mocked `getPairTrades` and you're re-implementing it's logic inside the mock.

Instead you should only mock the call to subgraph and test that action is dispatched by the actual implementation you're testing